src/monattack.cpp

```diff
@@ -1648,6 +1648,76 @@ void mattack::leap(monster *z, int index)
     }
 }
 
+void mattack::impale(monster *z, int index)
+{
+    if( !z->can_act() ) {
+        return;
+    }
+    Creature *target = z->attack_target();
+    if( target == nullptr || rl_dist( z->pos(), target->pos() ) > 1 ) {
+        return;
+    }
+
+    player *foe = dynamic_cast< player* >( target );
+    bool seen = g->u.sees( *z );
+
+    z->reset_special(index); // Reset timer
+    z->moves -= 80;
+    bool uncanny = foe != nullptr && foe->uncanny_dodge();
+    if( uncanny || dodge_check(z, target) ){
+    if( foe != nullptr ) {
+        if( seen ) {
+            auto msg_type = foe == &g->u ? m_warning : m_info;
+            foe->add_msg_player_or_npc( msg_type, _("The %s lunges at you, but you dodge!"),
+                                                _("The %s lunges at <npcname>, but they dodge!"),
+                                            z->name().c_str() );
+        }
+        if( !uncanny ) {
+            foe->practice( "dodge", z->type->melee_skill * 2 );
+            foe->ma_ondodge_effects();
+        }
+    } else if( seen ) {
+        add_msg( _("The %s lunges at %s, but misses!"), z->name().c_str(), target->disp_name().c_str() );
+    }
+    return;
+    }
+    auto hit = bp_torso;
+    int dam = rng(35, 45);
+    dam = target->deal_damage( z, hit, damage_instance( DT_CUT, dam ) ).total_damage();
+    if( dam > 0 && foe != nullptr ) {
+        if( seen ) {
+            auto msg_type = foe == &g->u ? m_bad : m_info;
+            //~ 1$s is monster name, 2$s bodypart in accusative
+            foe->add_msg_player_or_npc( msg_type,
+                                        _("The %1$s impales your torso!"),
+                                        _("The %1$s impales <npcname>'s torso!"),
+                                        z->name().c_str());
+        }
+        foe->practice( "dodge", z->type->melee_skill );
+        if( one_in( 30 / dam ) ) {
+            if( foe->has_effect( "infected", hit ) ) {
+                foe->add_effect( "infected", 50, hit, true );
+                foe->apply_damage( z, bp_torso, dam * 1.5 );
+                foe->moves -= dam * 20;
```

Comment:
Why does move drain only happen with the infection?
That damage scaling on infection is also really weird.

Plus, once again the whole tankiness thing: since it scales with damage, it means not being a kevlar-clad tank is a surefire way to get one-shot here.

If this part represents the zed ripping our own sharpened limbs out of the wound, you could have it deal some small amount of damage but with some arbitrarily huge amount of armor penetration (to represent the fact that is already in the wound rather than trying to get there).

Bleeding would be a better effect than infection here.